### ISSUE TITLE
Proposal Location Should Not Be Hardcoded skipped if Global

### ISSUE THREAD

**Issue by drewkg** [Azure/arm-ttk#506]
When deploying some resources the Location has to be Global, e.g. ActionGroups.  If deploying these resources alongside location specific resources inside the same Template there is no way to satisfy the Location Should Not Ne Hardcoded rules using the location parameter, resulting with:-

    [-] Location Should Not Be Hardcoded 156ms
      RuntimeException: azuredeploy.json must use the location parameter, not resourceGroup().location or deployment().location (except when used as a default value in the main template)

Proposal to pass this rule when a Location property is set to 'Global' as the rule does not make sense.

**Comment by bmoore-msft** [comment]
@drewkg - do you have a template snippet for a repro?  I'm not able to follow the scenarios with text alone ;)

**Comment by drewkg** [comment]
These two resource snippets would trigger the rule, as the Action Group has to have its location set to 'Global' While the Insights component is location dependant.

		{
			"type": "microsoft.insights/actionGroups",
			"apiVersion": "2019-03-01",
			"name": "[variables('actionGroupName')]",
			"location": "Global",
			"properties": {
				"groupShortName": "SmartDetect",
				"enabled": true,
				"emailReceivers": [],
				"smsReceivers": [],
				"webhookReceivers": [],
				"itsmReceivers": [],
				"azureAppPushReceivers": [],
				"automationRunbookReceivers": [],
				"voiceReceivers": [],
				"logicAppReceivers": [],
				"azureFunctionReceivers": []
			}
		},
		{
			"name": "[variables('applicationInsightsName')]",
			"type": "microsoft.insights/components",
			"apiVersion": "2018-05-01-preview",
			"location": "[parameters('resourceLocation')]",
			"tags": {
				"displayName": "Application Insights"
			},
			"kind": "web",
			"properties": {
				"Application_Type": "web",
				"Flow_Type": "Bluefield",
				"Request_Source": "rest",
				"RetentionInDays": 90,
				"publicNetworkAccessForIngestion": "Enabled",
				"publicNetworkAccessForQuery": "Enabled"
			}
		},

**Comment by bmoore-msft** [comment]
Thanks - I think the error is coming from the fact that you have a defaultValue on the parameter named "resourceLocation"?  Right now that's only allowed on a parameter if the name of the param is "location"...  Which I'm not sure is really needed - but as a breaking change should think about for a bit.

**Comment by drewkg** [comment]
I have adopted the new pattern of the parameter being called location, will change this template and re-validate if this is still an issue.

**Comment by drewkg** [comment]
@bmoore-msft You are correct, the issue was down to having 'resourceLocation' as the parameter name, once this was swapped over to just 'location' the test passed as expected.

Please close this issue as it is incorrect request.

Thank you for your time and response.
